grail/drand.py

```
@@ -1,147 +1,381 @@
+- Robust networking (Session, retries, shuffled relays, sensible timeouts)
+- Uniform schema for real & mock beacons
+"""
 
 import os
+import json
+import random
 import logging
 import requests
-from typing import Any, Dict, Optional
+from threading import Lock
+from __future__ import annotations
```

Ruff (0.12.2)

`from __future__` imports must occur at the beginning of the file

(F404)